spec/use_cases/provider_key_change_use_case_spec.rb

```diff
@@ -85,20 +107,53 @@ module Backend
         end
 
         it 'sets the default service for the new provider key' do
+          use_case.process
           expect(Service.default_id(new_key))
               .to eq service_ids_with_old_key.first.to_s
         end
 
         it 'removes (old) provider key data' do
+          use_case.process
           expect(Service.default_id(old_key)).to be nil
           expect(Service.list(old_key)).to be_empty
         end
 
         it 'changes the provider key associated with the affected services' do
+          use_case.process
           service_ids_with_old_key.each do |service_id|
             expect(Service.provider_key_for(service_id)).to eq new_key
           end
         end
+
+        it 'clears the Service cache keys of the old provider key' do
+          service_ids_with_old_key.each do |service_id|
+            exercise_provider_service_memoizer(old_key, service_id)
+            memoizer_keys = build_provider_service_memoizer_keys(old_key, service_id)
+
+            memoizer_keys.each do |memoized_key|
+              expect(Memoizer.memoized?(memoized_key)).to eq true
+            end
+            use_case.process
+            memoizer_keys.each do |memoized_key|
+              expect(Memoizer.memoized?(memoized_key)).to eq false
+            end
+          end
+        end
+
+        it 'clears the Service cache keys of the new provider key' do
+          service_ids_with_old_key.each do |service_id|
+            exercise_provider_service_memoizer(new_key, service_id)
+            memoizer_keys = build_provider_service_memoizer_keys(new_key, service_id)
+
+            memoizer_keys.each do |memoized_key|
```

Comment:
To test a change in the state of an object you can use the Rspec `change` matcher: https://relishapp.com/rspec/rspec-expectations/v/3-7/docs/built-in-matchers/change-matcher

In this case, we want to check how all the elements of an array changed, so using the `change` matcher becomes a bit more complex:
```
            expect { use_case.process }
                .to change { memoizer_keys.count { |k| Memoizer.memoized?(k) } }
                .from(memoizer_keys.size)
                .to(0)
```

I'm not proposing to make the change, just wanted to let you know that the matcher exists.

Reply:
Why not proposing it? Seems sensible.

Reply:
It's not only a matter of preference: it vastly simplifies the code, leaving less room for error, and it reads way better.

Reply:
I have changed it and now uses the change matcher